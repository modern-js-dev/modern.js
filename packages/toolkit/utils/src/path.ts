import path from 'path';
import upath from 'upath';

export const isRelativePath = (test: string): boolean =>
  /^\.\.?($|[\\/])/.test(test);

export const normalizeOutputPath = (s: string) => s.replace(/\\/g, '\\\\');
export const normalizeToPosixPath = (p: string) =>
  upath.normalizeSafe(path.normalize(p));
