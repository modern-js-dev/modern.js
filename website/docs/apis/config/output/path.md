---
sidebar_label: path
---

# `output.path`

:::info 适用的工程方案
* MWA
* 模块
:::

* 类型： `string`
* 默认值： `dist`


指定构建产物输出目录。

例如，可将构建产物输出目录改成 `build` 目录：

```javascript title="modern.config.js"
import { defineConfig } from '@modern-js/app-tools';

export default defineConfig({
  output: {
    path: 'build',
  },
});
```
