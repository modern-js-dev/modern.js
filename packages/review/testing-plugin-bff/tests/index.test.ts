import path from 'path';
import createPlugin, { setJestConfigForBFF } from '../src';

const root = path.resolve(__dirname, '../../../../');
expect.addSnapshotSerializer({
  test: val => typeof val === 'string' && val.includes(root),
  print: val =>
    typeof val === 'string'
      ? `"${val.replace(root, '').replace(/\\/g, '/')}"`
      : (val as string),
});

describe('testing-plugin-bff', () => {
  test('plugin', async () => {
    expect(createPlugin).toBeDefined();
    expect(createPlugin).toBeInstanceOf(Function);
  });

  test('setJestConfigForBFF', async () => {
    const appDir = path.normalize(path.resolve(__dirname, './fixtures/bff1'));
    const mockUtils = {
      _jestConfig: {},
      get jestConfig() {
        return this._jestConfig;
      },
      setJestConfig(config: any) {
        this._jestConfig = config;
      },
    };

    await setJestConfigForBFF({
      pwd: appDir,
      userConfig: {},
      plugins: [],
      routes: [],
      utils: mockUtils,
    });

    expect(mockUtils.jestConfig).toMatchSnapshot();
  });
});
