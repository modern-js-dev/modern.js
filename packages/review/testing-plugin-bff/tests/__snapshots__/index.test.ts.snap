// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing-plugin-bff setJestConfigForBFF 1`] = `
Object {
  "projects": Array [
    Object {
      "globals": Object {
        "modern_bff_info": Object {
          "appDir": "/packages/review/testing-plugin-bff/tests/fixtures/bff1",
          "modernUserConfig": Object {},
          "plugins": Array [],
          "routes": Array [],
        },
      },
      "moduleNameMapper": Object {},
      "resolver": "/packages/review/testing/src/config/resolver.ts",
      "rootDir": "/packages/review/testing-plugin-bff/tests/fixtures/bff1/api",
      "setupFilesAfterEnv": Array [
        "/packages/review/testing-plugin-bff/src/setup.ts",
      ],
      "testEnvironment": "node",
      "testMatch": Array [
        "**/api/**/*.test.[jt]s",
      ],
      "transform": undefined,
    },
  ],
}
`;

exports[`testing-plugin-bff testTimeout should not set in projects 1`] = `
Object {
  "projects": Array [
    Object {
      "globals": Object {
        "modern_bff_info": Object {
          "appDir": "/packages/review/testing-plugin-bff/tests/fixtures/bff1",
          "modernUserConfig": Object {},
          "plugins": Array [],
          "routes": Array [],
        },
      },
      "moduleNameMapper": Object {},
      "resolver": "/packages/review/testing/src/config/resolver.ts",
      "rootDir": "/packages/review/testing-plugin-bff/tests/fixtures/bff1/api",
      "setupFilesAfterEnv": Array [
        "/packages/review/testing-plugin-bff/src/setup.ts",
      ],
      "testEnvironment": "node",
      "testMatch": Array [
        "**/api/**/*.test.[jt]s",
      ],
      "transform": undefined,
    },
  ],
  "testTimeout": 1000,
}
`;
